File: app/src/main/java/at/shockbytes/corey/adapter/DaysScheduleAdapter.java
package at.shockbytes.corey.adapter;

import android.content.Context;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import org.joda.time.LocalDate;

import java.util.List;

import at.shockbytes.corey.R;
import at.shockbytes.corey.common.core.adapter.BaseAdapter;
import butterknife.Bind;

/**
 * @author Martin Macheiner
 *         Date: 02.12.2015.
 */
public class DaysScheduleAdapter extends BaseAdapter<String> {

    public DaysScheduleAdapter(Context context, List<String> data) {
        super(context, data);
    }

    @Override
    public BaseAdapter<String>.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        return new ViewHolder(inflater.inflate(R.layout.item_schedule_days, parent, false));
    }

    @Override
    public void onBindViewHolder(BaseAdapter.ViewHolder holder, int position) {
        String s = data.get(position) + "_"+position;
        holder.bind(s);
    }

    class ViewHolder extends BaseAdapter<String>.ViewHolder {

        @Bind(R.id.item_schedule_days_txt_name)
        TextView txtName;

        ViewHolder(final View itemView) {
            super(itemView);
        }

        @Override
        public void bind(String s) {

            String[] split = s.split("_");
            String item = split[0];
            int position = Integer.parseInt(split[1]);

            txtName.setText(item);

            if (LocalDate.now().getDayOfWeek()-1 == position) {
                txtName.setBackgroundResource(R.color.current_day);
            } else {
                txtName.setBackgroundResource(R.color.colorAccent);
            }
        }
    }

}

File: app/src/main/java/at/shockbytes/corey/ui/adapter/DaysScheduleAdapter.kt
package at.shockbytes.corey.ui.adapter

import android.content.Context
import androidx.core.content.ContextCompat
import android.view.View
import android.view.ViewGroup
import at.shockbytes.corey.R
import at.shockbytes.util.adapter.BaseAdapter
import kotlinx.android.extensions.LayoutContainer
import kotlinx.android.synthetic.main.item_schedule_days.*
import org.joda.time.LocalDate

/**
 * Author:  Martin Macheiner
 * Date:    02.12.2015
 */
class DaysScheduleAdapter(
    context: Context,
    data: List<String>,
    onItemLongClickListener: OnItemLongClickListener<String>
) : BaseAdapter<String>(
    context,
    onItemLongClickListener = onItemLongClickListener
) {

    init {
        this.data.apply {
            clear()
            addAll(data)
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): BaseAdapter.ViewHolder<String> {
        return ViewHolder(inflater.inflate(R.layout.item_schedule_days, parent, false))
    }

    private inner class ViewHolder(
        override val containerView: View
    ) : BaseAdapter.ViewHolder<String>(containerView), LayoutContainer {

        override fun bindToView(content: String, position: Int) {

            val bgColor: Int
            val txtColor: Int
            if (LocalDate.now().dayOfWeek - 1 == position) {
                bgColor = ContextCompat.getColor(context, R.color.colorAccent)
                txtColor = ContextCompat.getColor(context, R.color.colorAccent)
            } else {
                bgColor = ContextCompat.getColor(context, R.color.white)
                txtColor = ContextCompat.getColor(context, R.color.colorPrimaryText)
            }

            card_item_schedule_days.strokeColor = bgColor
            item_schedule_days_txt_name.apply {
                text = content
                setTextColor(txtColor)
            }
        }
    }
}